File: handler.js
'use strict';
require('dotenv').config();
const nodemailer = require('nodemailer');

module.exports.sendEmail = async (event) => {
  try {
    const { receiver_email, subject, body_text } = JSON.parse(event.body || '{}');

    if (!receiver_email || !subject || !body_text) {
      return {
        statusCode: 400,
        body: JSON.stringify({ success: false, message: 'Missing required fields.' }),
      };
    }

    // Create transporter (Gmail SMTP example)
    let transporter = nodemailer.createTransport({
      host: process.env.SMTP_HOST,
      port: process.env.SMTP_PORT,
      secure: false,
      auth: {
        user: process.env.SMTP_USER,
        pass: process.env.SMTP_PASS,
      },
    });

    await transporter.sendMail({
      from: `"Serverless Email API" <${process.env.SMTP_USER}>`,
      to: receiver_email,
      subject: subject,
      text: body_text,
    });

    return {
      statusCode: 200,
      body: JSON.stringify({ success: true, message: 'Email sent successfully!' }),
    };

  } catch (err) {
    console.error('SMTP Error:', err);
    console.log('SMTP Configuration:', {
      host: process.env.SMTP_HOST,
      port: process.env.SMTP_PORT,
      user: process.env.SMTP_USER ? 'Set' : 'Not Set',
      pass: process.env.SMTP_PASS ? 'Set' : 'Not Set'
    });
    
    return {
      statusCode: 500,
      body: JSON.stringify({ 
        success: false, 
        message: 'Failed to send email', 
        error: err.message,
        details: err.response || 'No additional error details'
      }),
    };
  }
};

File: handler.py
import json
import smtplib
from email.mime.text import MIMEText

def send_email(event, context):
    try:
        body = json.loads(event["body"])
        receiver_email = body.get("receiver_email")
        subject = body.get("subject")
        body_text = body.get("body_text")

        if not receiver_email or not subject or not body_text:
            return {
                "statusCode": 400,
                "body": json.dumps({"success": False, "error": "Missing required fields"})
            }

        # Gmail SMTP setup - Replace with your Gmail and App Password
        sender_email = "shivarajkumarbm21@gmail.com"
        sender_password = "ujza llyz zgoy cgud"  # Use Gmail App Password, not your regular password

        msg = MIMEText(body_text)
        msg["Subject"] = subject
        msg["From"] = f"Serverless API"
        msg["To"] = receiver_email

        with smtplib.SMTP_SSL("smtp.gmail.com", 465) as server:
            server.login(sender_email, sender_password)
            server.send_message(msg)

        return {
            "statusCode": 200,
            "body": json.dumps({"success": True, "message": "Email sent successfully!"})
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"success": False, "error": str(e)})
        }
